### PR TITLE
fix clean state zk

### PR DESCRIPTION
**Pull request by huijunw**
fix a bug in runtime mgr cleanState()
to best effort remove all the state

### CODE REVIEW

File: heron/scheduler-core/src/java/com/twitter/heron/scheduler/RuntimeManagerRunner.java [hunk 1]
```diff
@@ -209,67 +209,77 @@ void updateTopologyHandler(String topologyName, String newParallelism)
 
   /**
    * Clean all states of a heron topology
+   * pre-condition: all containers are gone
    * 1. Topology def and ExecutionState are required to exist to delete
    * 2. TMasterLocation, SchedulerLocation and PhysicalPlan may not exist to delete
+   * post-condition: best effort all states are gone, otherwise then exception
```
Comment (nwangtw) [inline heron/scheduler-core/src/java/com/twitter/heron/scheduler/RuntimeManagerRunner.java]: s/then/throw?

File: heron/scheduler-core/src/java/com/twitter/heron/scheduler/RuntimeManagerRunner.java [hunk 2]
```diff
@@ -209,67 +209,77 @@ void updateTopologyHandler(String topologyName, String newParallelism)
...
     LOG.fine("Cleaning up topology state");
 
-    Boolean result;
-
     // It is possible that TMasterLocation, MetricsCacheLocation, PackingPlan, PhysicalPlan and
     // SchedulerLocation are not set. Just log but don't consider it a failure and don't return
     // false
-    result = statemgr.deleteTMasterLocation(topologyName);
-    if (result == null || !result) {
+    Boolean result1 = statemgr.deleteTMasterLocation(topologyName);
```
Comment (nwangtw) [inline heron/scheduler-core/src/java/com/twitter/heron/scheduler/RuntimeManagerRunner.java]: real names for the results.

File: heron/statemgrs/src/java/com/twitter/heron/statemgr/zookeeper/curator/CuratorStateManager.java [hunk 3]
```diff
@@ -273,6 +273,7 @@ public String getConnectionString() {
 
     } catch (KeeperException e) {
       if (KeeperException.Code.NONODE.equals(e.code())) {
+        LOG.info("Node does not exist, delete success " + path);
```
Comment (nwangtw) [inline heron/statemgrs/src/java/com/twitter/heron/statemgr/zookeeper/curator/CuratorStateManager.java]: s/delete/considered as success

**Comment by huijunw** [comment]
after careful examine, i found this pr is not necessary.

**Comment by srkukarni** [comment]
@huijunw Can you please comment why this was not found necessary?

**Pull request closed without merging**